Mathlib/Topology/Order/Bornology.lean

```diff
@@ -0,0 +1,111 @@
+-/
+import Mathlib.Topology.Bornology.Constructions
+
+/-!
+# Bornology of order-bounded sets
+
+This file relates the notion of bornology-boundedness (sets that lie in a bornology) to the notion
+of order-boundedness (sets that are bounded above and below).
+
+## Main declarations
+
+* `orderBornology`: The bornology of order-bounded sets of a nonempty lattice.
+* `IsOrderBornology`: Typeclass predicate for a preorder to be equipped with its order-bornology.
+-/
+
+open Bornology Set
+
+variable {α : Type*} [Bornology α] {s t : Set α}
+
+section Lattice
+variable [Lattice α] [Nonempty α]
+
+/-- Order-bornology on a nonempty lattice. The bounded sets are the sets that are bounded both above
+and below. -/
+def orderBornology : Bornology α := .ofBounded
+  {s | BddBelow s ∧ BddAbove s}
+  (by simp)
+  (fun s hs t hst ↦ ⟨hs.1.mono hst, hs.2.mono hst⟩)
+  (fun s hs t ht ↦ ⟨hs.1.union ht.1, hs.2.union ht.2⟩)
+  (by simp)
+
+@[simp] lemma orderBornology_isBounded : orderBornology.IsBounded s ↔ BddBelow s ∧ BddAbove s := by
+  simp [IsBounded, IsCobounded, -isCobounded_compl_iff]
+
+end Lattice
+
+variable (α) [Preorder α] in
+/-- Predicate for a preorder to be equipped with its order-bornology, namely for its bounded sets
+to be the ones that are bounded both above and below. -/
+class IsOrderBornology : Prop where
+  protected isBounded_iff_bddBelow_bddAbove (s : Set α) : IsBounded s ↔ BddBelow s ∧ BddAbove s
+
+lemma isOrderBornology_iff_eq_orderBornology [Lattice α] [Nonempty α] :
+    IsOrderBornology α ↔ ‹Bornology α› = orderBornology := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ⟨fun s ↦ by rw [h, orderBornology_isBounded]⟩⟩
+  ext s
+  exact isBounded_compl_iff.symm.trans (h.1 _)
+
+section Preorder
+variable [Preorder α] [IsOrderBornology α]
+
+lemma isBounded_iff_bddBelow_bddAbove : IsBounded s ↔ BddBelow s ∧ BddAbove s :=
+  IsOrderBornology.isBounded_iff_bddBelow_bddAbove _
+#align bounded_iff_bdd_below_bdd_above isBounded_iff_bddBelow_bddAbove
+#align real.bounded_iff_bdd_below_bdd_above isBounded_iff_bddBelow_bddAbove
+
+protected lemma Bornology.IsBounded.bddBelow (hs : IsBounded s) : BddBelow s :=
+  (isBounded_iff_bddBelow_bddAbove.1 hs).1
+#align metric.bounded.bdd_below Bornology.IsBounded.bddBelow
+
+protected lemma Bornology.IsBounded.bddAbove (hs : IsBounded s) : BddAbove s :=
+  (isBounded_iff_bddBelow_bddAbove.1 hs).2
+#align metric.bounded.bdd_above Bornology.IsBounded.bddAbove
+
+protected lemma BddBelow.isBounded (hs₀ : BddBelow s) (hs₁ : BddAbove s) : IsBounded s :=
+  isBounded_iff_bddBelow_bddAbove.2 ⟨hs₀, hs₁⟩
+#align bdd_below.bounded BddBelow.isBounded
+
+protected lemma BddAbove.isBounded (hs₀ : BddAbove s) (hs₁ : BddBelow s) : IsBounded s :=
+  isBounded_iff_bddBelow_bddAbove.2 ⟨hs₁, hs₀⟩
+#align bdd_above.bounded BddAbove.isBounded
+
+lemma BddBelow.isBounded_inter (hs : BddBelow s) (ht : BddAbove t) : IsBounded (s ∩ t) :=
+  (hs.mono $ inter_subset_left _ _).isBounded $ ht.mono $ inter_subset_right _ _
```

Comment:
I think you might have another conflict with yourself on `inter_subset_left`

Reply:
Well spotted